### ISSUE TITLE
Training with single GPU

### ISSUE THREAD

**Issue by kristellmarisse** [weiliu89/caffe#96]
The default SSD training is using 4 gpus. Since I have only one GPU (GTX960 4gb), I have modified the python script to make it run using single GPU. I am able to train on VOC dataset and my own custom dataset as well. So by changing the number of GPUs should I automatically scale the number of iterations and step size accordingly? @weiliu89 

Thanks !

**Comment by weiliu89** [comment]
If you keep accum_batch to 32, you do not need to change other things except the batch_size. (Make sure you have pulled the latest python script, there was a bug in adjusting the learning rate if batch_size != accum_batch).

**Comment by nbadalls** [comment]
I use only GPU(GTX970 4gb) to train model on VOC dataset too. I change GPU number from "0,1,2,3 " to "0" and set batch_size = 8 , accum_batch_size = 32. After 50 iterations, loss suddenly changes to nan from 7.02633. I don't know how  to deal with it. Did you encounter such weird situation? @kristellmarisse

**Comment by weiliu89** [comment]
@nbadalls You should probably try to lower the learning rate in the beginning. And then resume the training with original learning rate.

**Comment by nbadalls** [comment]
Thanks for your answer.@weiliu89 
I try to lower the leaning rate to 1e-4, 1e-5 or 1e-7 in the beginning , then recover the learning rate back to 1e-3 and start the training. But finally the loss turns to nan again.It seems not work.

**Comment by weiliu89** [comment]
You have to train for a couple hundred or thousand iterations to make sure training is stable, and then resume training with larger learning rate.

**Comment by nbadalls** [comment]
I see. But the loss  changes to nan before thousands iterations when I change the learning rate to 1e-4, 1e-5 or 1e-7 even a more extreme condition — zeros.  The lower learning rate I use, there are more number of iterations before the loss changes to nan.
